### PR TITLE
Only apply addlDelta for IMAGE_REL_BASED_REL32 relocs

### LINKED ISSUE #79170: JIT.Regression\JitBlue\Runtime_31615\Runtime_31615\Runtime_31615 fails
```
      > C:\h\w\A3A508C1\w\B8E60A0B\e\JIT\Regression\JitBlue\Runtime_31615\Runtime_31615\Runtime_31615.cmd
      Expected: True
      Actual:   False
      Stack Trace:
           at JIT_Regression._JitBlue_Runtime_31615_Runtime_31615_Runtime_31615_._JitBlue_Runtime_31615_Runtime_31615_Runtime_31615_cmd()
           at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
           at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
      Output:
        
        Return code:      1
```
Failed in [many](https://runfo.azurewebsites.net/search/tests/?q=started%3A%7E5+name%3A%22Runtime_31615%22) PRs

### PR DESCRIPTION
**Pull request by BruceForstall**
This matches what ngen did, and makes sense since only IP-relative relocations need to addlDelta to determine the address of the next instruction.

Fixes #79170

### CODE REVIEW

File: src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs [hunk 1]
```diff
@@ -3752,10 +3752,14 @@ private void recordRelocation(void* location, void* locationRW, void* target, us
                     break;
             }
 
-            relocDelta += addlDelta;
-
             TargetArchitecture targetArchitecture = _compilation.TypeSystemContext.Target.Architecture;
             RelocType relocType = GetRelocType(targetArchitecture, fRelocType);
+
+            if (relocType == RelocType.IMAGE_REL_BASED_REL32)
```
Comment (jkotas) [inline src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs:3758]: Would it make more sense for the JIT to pass `addlDelta` as zero? 

One can imagine cases where `addlDelta` can be useful for other relocation types as well. They may not exist today, but they can show up in future.

Comment (BruceForstall) [reply src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs:3758]: Maybe? That's what https://github.com/dotnet/runtime/pull/79467 does. I don't have a strong opinion either way.

I was interested to learn when I investigated that ngen ignored addlDelta except for REL32, and I'm generally wary of ways in which the crossgen2 implementation of the JIT/EE interface diverges from the ngen implementation.

Comment (jkotas) [reply src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs:3758]: I think #79467 is a better fix.

I would not use ngen or .NET Framework as a gold standard for JIT/EE interface implementation. The JIT/EE interface contract in .NET Framework had a lot of things in it that did not make sense and it was a lot of friction to fix them.

**Pull request closed without merging**